stdlib/result.mli

```diff
@@ -0,0 +1,89 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*   Copyright 2018 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Result values.
+
+    @since 4.08 *)
```

Comment:
This could probably be a bit more explanatory. How about

> `Result` values.
> The `Result` type is much like the `Option` type, but it allows an explicit error value to be carried instead of `None`.

Reply:
> `Result` values.
>
> `Result` is the canonical coproduct.

Reply:
As I've said elsewhere, the purpose of documentation is often to guide beginners through understanding a facility. No beginner will know what the "canonical coproduct" means.

Reply:
I think the new text is better, but perhaps a slight tweak to "Result values _are used to return_ computation results in an explicit and declarative manner..." would be better. Then it is obvious to a beginner that this is used as a return type to wrap results.